### ISSUE TITLE
ui: session - clicking outside the Session Timeout Warning Dialog closes the dialog without handling "Stay connected" 

### ISSUE THREAD

**Issue by SuperITMan** [NationalBankBelgium/stark#1561]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#got-a-question-or-problem
</code></pre>

## Current behavior

When the Session Timeout Warning Dialog appears 

![image](https://user-images.githubusercontent.com/10333565/72804432-3dc73a00-3c48-11ea-9933-d8d1f8c9c482.png)

And you click outside the dialog, it closes the dialog without handling the "Stay connected".

**EDIT**: If you click outside of the dialog, it closes it and does not stop the countdown. That means that you are disconnected once the countdown is finished. That's really bad for the UX :confused: 

## Expected behavior

2 behaviours are possible here:
  1. When clicking outside the dialog, nothing happens
  2. When clicking outside the dialog, the "Stay connected" behaviour is triggered. 

## Minimal reproduction of the problem with instructions

Open the showcase, wait for the Session Timeout Warning Dialog appearing then click outside the dialog.
## Environment

<pre><code>
Angular version: 7.x
Stark version: 10.0.0-rc.3

</code></pre>

**Comment by SuperITMan** [comment]
@nicanac @christophercr Which behaviour do you prefer ?

Personally I prefer the second one, clicking outside should make the dialog disappear.

**Comment by christophercr** [comment]
Mmm I think I prefer option 1: clicking outside does nothing. That way the user is forced to click on the "Stay connected" button if he really wishes to keep his session alive (that also feels like kind of safer right?).

Otherwise, triggering the "stay connected" behavior when clicking outside might be confusing... because if the app displays such dialog it is for a reason and the user is expected to interact with it

**Comment by nicanac** [comment]
At first, option 2 was my favorite because in most case like in a picture viewer, when you click outside it closes the popup.
But, I agree with you @christophercr , we are in the case of a connection, a bit like if it was a login.
To avoid any confusion, I've the feeling we have to force the user to at least read the button :-)

**Comment by SuperITMan** [comment]
Okok. I agree.
In that case, if the user wants to close his session at that time, how can he do it ? :blush:
I suggest to add a button "Disconnect". What do you think ?

**Comment by nicanac** [comment]
> Okok. I agree.
> In that case, if the user wants to close his session at that time, how can he do it ? 😊
> I suggest to add a button "Disconnect". What do you think ?

Yes, good idea.

**Comment by christophercr** [comment]
Yep, good idea... but is it really something urgently requested by anyone?

**Comment by SuperITMan** [comment]
@christophercr
Actually, it's not a matter of implementing a new feature but of fixing a bug.
I edited my issue to explain better the current behaviour which is bad.

**Comment by christophercr** [comment]
Ah ok, well, in that case the solution is simple: just add the option `disableClose` in this line: https://github.com/NationalBankBelgium/stark/blob/aa9d3e440e49253651aa0c6648a6f1c151eefb5a/packages/stark-ui/src/modules/session-ui/effects/session-timeout-warning.effects.ts#L42

And about the `Disconnect` button, I don't think is really necessary since the user will be logged out anyways (just some X seconds later). That said, I'm not against about such "nice to have" either ;)

**Comment by SuperITMan** [comment]
Yep'! Exactly. It's a simple fix :-) (that should be part of 10.0.0-rc.4 I think)
Maybe we can create another issue/PR for the "Disconnect" button. What do you think ? :blush:

**Comment by christophercr** [comment]
Alright! Then let's create another issue/PR for it 😉

**Comment by christophercr** [comment]
I've created issue #1731 for the implementation of the "Disconnect" button 😉

**Comment by christophercr** [comment]
Closed by #1605
